plugin/my_asset_manager/MyAssetManagerInterface.py

```diff
@@ -121,6 +120,95 @@ def isEntityReferenceString(self, someString, hostSession):
+        # a write access mode via calling the error callback.
+        if entityTraitsAccess != EntityTraitsAccess.kRead:
+            result = BatchElementError(
+                BatchElementError.ErrorCode.kEntityAccessError, "Entities are read-only"
+            )
+            for idx in range(len(entityReferences)):
+                errorCallback(idx, result)
+            return
+
+        # Iterate over all the entity references, calling the correct
+        # error/success callbacks into the host.
+        # You should handle success/failure on an entity-by-entity
+        # basis, do not abort your entire operation because any single
+        # entity is malformed/can't be processed for any reason, use
+        # the error callback and continue.
+        for idx, ref in enumerate(entityReferences):
+            # It may be that one of the references you are provided is
+            # recognized for this manager, but has some syntax error or
+            # is otherwise incorrect for your specific resolve context.
+            # For example, an asset reference that specifies a version
+            # for an un-versioned entity could be considered malformed.
+            #
+            # N.B. It's not required to perform an explicit check here
+            # if this is naturally serviced during your backend lookup,
+            # the key is not to error the whole batch, but use the error
+            # callback for relevant references.
+            identifier_is_malformed = is_malformed_ref(ref)
+            if identifier_is_malformed:
+                error_result = BatchElementError(
+                    BatchElementError.ErrorCode.kMalformedEntityReference,
+                    "Entity identifier is malformed",
+                )
+                errorCallback(idx, error_result)
+            else:
+                # If our manager has the asset in question, we can
+                # let the host know which traits make up this specific
+                # entity.
+                if ref.toString() in managed_filesystem_locations:
+                    # This manager only manages location, and manages it
+                    # for every entity, so we can just build a success
+                    # result consisting of Entity + LocatableContent. It
+                    # is likely you will want to replace this with a
+                    # query into your backend system to inspect specific
+                    # entities.
+                    success_result = {EntityTrait.kId, LocatableContentTrait.kId}
+                    if ref.toString() == "my_asset_manager:///anAsset2":
+                        # Arbitrarily presume that the second asset is
+                        # a configuration entity.
+                        # In reality you would be determining this based
+                        # on your backend system.
+                        success_result.add(ConfigTrait.kId)
```

Comment:
[consider][minor] This is fine for exemplification, though the trait set mapping is kinda buried deep in the code. Perhaps the (unused, see other comment) `managed_filesystem_locations` could be adapted to `entity_traitset_map`, or something?

Reply:
yeah this is a pretty neat idea, i'll do that.

Reply:
https://github.com/OpenAssetIO/Template-OpenAssetIO-Manager-Python/pull/19/commits/d14e9d3bae80e53b33cb15a97f6687046fe19245

I feel this removes the need for a fair bit of the doc, because it's apparently that the traits are mapped to the entities now from the example.

Reply:
Yeah I like it.